### PR TITLE
[Term Entry] C++ Math Functions: `pow()`

### LINKED ISSUE #959: [Term Entry] C++ Math Functions: `pow()`
### Reason/inspiration (optional)

Under `cpp/concepts/math-functions/terms/`, create a new term entry for the `pow()` function.

- In the `terms/` folder, make a new folder called `pow`
- Inside the `pow` folder, create a new markdown file called `pow.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Let's include an "Example" section with a \```cpp code snippet.
- Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Libens-bufo**
### Description

Closes #959 

Created `pow.md` in `/terms/pow/` and authored relevant entry.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:47]: Both examples are largely the same. It'd be better if you did something to differentiate the two. Maybe having one use fractional exponents to find the roots of a number:

double base = 27;
double root = 3;
double exponent = 1 / root;

That's just a suggestion, you could also do something like finding the volume of a cube or a sphere.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Libens-bufo, I did a first review and left you a few comments.

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:38]:
```suggestion
```

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (SSwiniarski) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:59]:
```suggestion
```

**Comment by Libens-bufo** [comment]
I have made the suggested changes, have requested another review, and would like your thoughts on the new example and output string.

**Comment by SSwiniarski** [comment]
> I have made the suggested changes, have requested another review, and would like your thoughts on the new example and output string.

@Libens-bufo, I think the second example looks good.

**Comment by Libens-bufo** [comment]
Checks are good now (did `yarn format`).

Requesting second review. @Dusch4593

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:14]:
```suggestion
The **`pow()`** function returns the result of a base value raised to the power of an exponent.
```

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:21]: Can we put some text below the syntax snippet that describes the domain and range of `base` and `exponent` as well as what errors are caused by which invalid input(s)?

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:39]:
```suggestion
    //Output: 2 raised to the power of 2 equals 4.
```

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:60]:
```suggestion
    std::cout << base << " raised to the power of " << exponent << " equals " << result <<", which is the " << root << "-root of " << base << ".\n";
```

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:45]: This example seems too similar to the [example](https://www.codecademy.com/resources/docs/cpp/math-functions/cbrt#:~:text=Use%20cbrt()%20to%20return%20the%20cube%20root%20of%2027%3A) under the existing `cbrt()` C++ entry. Wondering if we could change the values but keep the reciprocal?

```suggestion
The following example is runnable and uses the `pow()` function to find the nth-root of `16` by raising the base to the reciprocal of the desired root:
```

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:54]:
```suggestion
    double base = 16;
    double root = 4;
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@Libens-bufo Finished with my review and left you some comments/suggestions. 😄

**Comment by Dusch4593** [comment]
Hey @Libens-bufo! Just wanted to check in on things with this PR? Any questions or anything we can do to help?

**Comment by Libens-bufo** [comment]
Apologies, been swamped. Will submit requested changes this week. From: Brandon Dusch ***@***.***>Sent: Monday, September 26, 2022 3:01 PMTo: Codecademy/docs ***@***.***>Cc: Jason Swinehart ***@***.***>; Mention ***@***.***>Subject: Re: [Codecademy/docs] [Term Entry] C++ Math Functions: `pow()` (PR #1000) 
Hey @Libens-bufo! Just wanted to check in on things with this PR? Any questions or anything we can do to help?

**Comment by yangc95** [comment]
Hey @Libens-bufo, did you need some assistance with these open comments?

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (yangc95) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:57]:
```suggestion
  double base = 16;
```

**Review by SSwiniarski — APPROVED** [review]
@yangc95 Looks good to me.

File: content/cpp/concepts/math-functions/terms/pow/pow.md
Comment (yangc95) [inline content/cpp/concepts/math-functions/terms/pow/pow.md:26]:
```suggestion
- A range error may occur due to underflow.
```

**Review by yangc95 — APPROVED** [review]

**Review by Dusch4593 — APPROVED** [review]
cc: @yangc95 @SSwiniarski

@Libens-bufo LGTM! 👍🏻

Made a few small edits in 93fdb16746fb8fe12e70b35a3bb226d9d6103b45 but this looks good to merge now. 🚀

**Comment by Dusch4593** [comment]
Congrats and thank you so much for contributing to Codecademy Docs, @Libens-bufo!! 🚀 🙌🏻 🥳

The entry should now be live: https://www.codecademy.com/resources/docs/cpp/math-functions/pow

Currently, your Codecademy profile anonymously listed on the live entry. To have it listed, you'll need to add your GitHub username to your Codecademy profile, log out of Codecademy, and log back in through your GitHub account. Let us know if you'd like any help with this:

Screen Shot 2022-10-31 at 1 01 04 PM

Looking forward to your next contribution to Docs! 😄

**Pull request merged**
